src/tests/mldsa.rs

```diff
@@ -1090,3 +1090,94 @@ fn test_mldsa_operations() {
     ));
     assert_eq!(out, signature);
 }
+
+#[test]
+#[parallel]
+fn test_mldsa_seedonly() {
+    let mut testtokn = TestToken::initialized("test_mldsa_seedonly", None);
+
+    let session = testtokn.get_session(true);
+
+    /* login */
+    testtokn.login();
+
+    let (_, privkey) = ret_or_panic!(generate_key_pair(
+        session,
+        CKM_ML_DSA_KEY_PAIR_GEN,
+        &[
+            (CKA_CLASS, CKO_PUBLIC_KEY),
+            (CKA_PARAMETER_SET, CKP_ML_DSA_44),
+            (CKA_KEY_TYPE, CKK_ML_DSA),
+        ],
+        &[],
+        &[(CKA_VERIFY, true)],
+        &[(CKA_CLASS, CKO_PRIVATE_KEY), (CKA_KEY_TYPE, CKK_ML_DSA)],
+        &[(CKA_LABEL, format!("ML-DSA private key 1").as_bytes()),],
+        &[
+            (CKA_SIGN, true),
+            (CKA_SENSITIVE, false),
+            (CKA_EXTRACTABLE, true)
+        ],
+    ));
+
+    /* extract the private seed, and create a new private key with the seed only.
+     * Test that it has a private value and it matches the original key's one */
+    let mut extract_template = make_ptrs_template(&[
+        (CKA_SEED, std::ptr::null_mut(), 0),
+        (CKA_VALUE, std::ptr::null_mut(), 0),
+    ]);
+    let ret = fn_get_attribute_value(
+        session,
+        privkey,
+        extract_template.as_mut_ptr(),
+        extract_template.len() as CK_ULONG,
+    );
+    assert_eq!(ret, CKR_OK);
+
+    let mut seed = vec![0u8; extract_template[0].ulValueLen as usize];
+    extract_template[0].pValue = seed.as_mut_ptr() as CK_VOID_PTR;
+    let mut value = vec![0u8; extract_template[1].ulValueLen as usize];
+    extract_template[1].pValue = value.as_mut_ptr() as CK_VOID_PTR;
+
+    let ret = fn_get_attribute_value(
+        session,
+        privkey,
+        extract_template.as_mut_ptr(),
+        extract_template.len() as CK_ULONG,
+    );
+    assert_eq!(ret, CKR_OK);
+
+    let privkey2 = ret_or_panic!(import_object(
+        session,
+        CKO_PRIVATE_KEY,
+        &[
+            (CKA_KEY_TYPE, CKK_ML_DSA),
+            (CKA_PARAMETER_SET, CKP_ML_DSA_44),
+        ],
+        &[
+            (CKA_SEED, &seed),
+            (CKA_LABEL, format!("ML-DSA private key 2").as_bytes()),
+        ],
+        &[
+            (CKA_SIGN, true),
+            (CKA_SENSITIVE, false),
+            (CKA_EXTRACTABLE, true)
+        ],
+    ));
+
+    let mut value2 = vec![0u8; value.len()];
+    let mut value2_template = make_ptrs_template(&[(
+        CKA_VALUE,
+        value2.as_mut_ptr() as CK_VOID_PTR,
+        value2.len(),
+    )]);
+    let ret = fn_get_attribute_value(
+        session,
+        privkey2,
+        value2_template.as_mut_ptr(),
+        value2_template.len() as CK_ULONG,
+    );
```

Comment:
could we reuse the `extract_key_value()` or `check_attributes()` to avoid a lot of boilerplate? Same for the mlkem test.